common/proto/dex_amm.proto

```diff
@@ -6,22 +6,135 @@ import "erc20.proto";
 package messari.dex_amm.v1;
 
 message Pool {
```

Comment:
I think one way to design this is to have a common `Pool` struct across all protocols, with fields like TVL. Each protocol type would extend the `Pool` struct to include more protocol-specific fields. This way we can have shared methods like `SumTVL` that would take an array of `Pools` and return the total protocol TVL, and have it be applicable for all protocols

Reply:
Yeah I like that idea, I think we should do it.